### DISCUSSION TITLE
会考虑增加maosn等插件的教程吗

### DISCUSSION THREAD

**Discussion by LionChenA** [nshen/learn-neovim-lua#119] [General]
插件[lsp-installer](https://github.com/williamboman/nvim-lsp-installer)的作者[williamboman](https://github.com/williamboman)已经开发出了lsp-installer 的下一代插件[mason.nvim](https://github.com/williamboman/mason.nvim) 并不再维护原插件

**Comment by nshen** [comment]
~~还没有换，我的 lsp-installer 好像还是可以正常的安装和更新 language server，再加上我也没什么新需求，稳定了也挺好的。~~

~~mason更新挺频繁，怕有bug，所以暂时还没折腾，懒。~~

~~不知道有没有大佬分享一下使用体验。~~

更新升级 mason 教程 https://juejin.cn/post/7154005621887631396

**Comment by ghost** [comment]
mason 的话，这两天刚玩完，加上下面这段代码就能用了。servers 改成你需要的就可以了。

```lua
local status, mason = pcall(require, "mason")
if (not status) then return end
local status2, lspconfig = pcall(require, "mason-lspconfig")
if (not status2) then return end

mason.setup({})

local servers = { "bashls", "jsonls", "sumneko_lua", "gopls", "solidity", "yamlls" }
lspconfig.setup {
    ensure_installed = servers,
}

local nvim_lsp = require "lspconfig"

lspconfig.setup_handlers {
    function(server_name)
        nvim_lsp[server_name].setup{}
    end,
}
```

**Comment by nshen** [comment]
我也升级 mason 了 https://juejin.cn/post/7154005621887631396
